src/services/ogmios/processBlock.ts

```diff
@@ -28,41 +30,101 @@ const buildPersonalization = async (metadata: PersonalizationOnChainMetadata): P
     return personalization;
 };
 
+function isValidDatum(datumObject: PersonalizationOnChainMetadata): boolean {
+    // TODO: validate datum
+    return true;
+}
+
 const processAssetReferenceToken = async ({
     assetName,
     slotNumber,
     datum
 }: {
     assetName: string;
     slotNumber: number;
-    datum:
-        | string
-        | {
-              [k: string]: unknown;
-          }
-        | null
-        | undefined;
+    datum?: string;
 }) => {
-    const hexName = assetName?.split(MetadatumAssetLabel.SUB_STANDARD_NFT)[1];
+    const hexName = assetName?.split(MetadatumAssetLabel.REFERENCE_NFT)[1];
     if (!hexName) {
-        Logger.log(`unable to decode ${hexName}`);
+        Logger.log(`unable to decode reference token name: ${hexName}`);
+        return;
+    }
+
+    const name = Buffer.from(hexName, 'hex').toString('utf8');
+
+    if (!datum) {
+        // our reference token should always have datum.
+        // If we do not have datum, something is wrong.
+        // TODO: what happens during a token burn?
```

Comment:
There is probably datum, but we don't care because we're burning

Reply:
Cool! I have a ticket for burning so I can tackle it once I get there.